docker/Dockerfile

```diff
@@ -0,0 +1,47 @@
+ARG UBUNTU_VERSION=18.04
+FROM ubuntu:${UBUNTU_VERSION}
+
+# Use bash as the default shell
+RUN ln -snf /bin/bash /bin/sh
+
+# Update core packages
+RUN apt-get -y update
+RUN apt-get -y upgrade
+RUN apt-get -y dist-upgrade
+
+# Get some apt packages
+RUN apt-get -y install autoconf clang cmake g++ gcc gfortran git libblas-dev \
+                       libhdf5-dev liblapack-dev libpython2.7-dev libtool \
+                       python-numpy python-pip python-setuptools wget
+
+# Get astyle
+RUN if [ "${UBUNTU_VERSION}" == "18.04" ]; then \
```

Comment:
Optional: I would recommend just having the `18.04` condition here.  Dockerfile specifications tend to be operating system-version specific.  It is probably better to fail fast here instead of trying to compensate.